chef/automate
# Command to create the organisations in chef server

### :nut_and_bolt: Description: What code changed, and why?
Command to add the sample data of organisations in chef server using web ui key
### :chains: Related Resources
https://github.com/chef/automate/issues/6532
### :+1: Definition of Done
I have added changes to load the sample data of organisations in chef server
Using command we can add the sample organisations in server on chef server side using web ui key.
For script, added an API to create the organisations on chef server.
API reads the web ui key of given chef server and make request to create an organisations.

### :athletic_shoe: How to Build and Test the Change

1. Command to add the orgs in server
     You need to enter the hab studio then check the services are running.
     To start the services you can use ```start_infra_proxy_service``` or ```start_all_services```
     Then run ```infra_service_load_sample_orgs```

    Options:
    -N: Number of records, default 50
    -S: Chef infra server id, which should be available on chef server and added in automate
    -O: Chef organization's name prefix, default: chef-org

    Example:
    ```infra_service_load_sample_orgs -N 50 -S local-dev -O chef-org ```

2. API to create an orgs in chef server
    Steps to build:
    ```
    rebuild components/infra-proxy-service
    rebuild components/automate-gateway
    ```
    Steps to test:

    1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
    2. Add the chef-server test environment credentials and extract the SERVER IDs
    3. Add the organisation name and full name in the request.

    Request
    ```
    curl --location --request POST 'https://a2-dev.test/api/v0/infra/servers/{SERVERID}/infraserverorgs' \
    --header 'api-token: your admin token' \
    --header 'Content-Type: application/json' \
    --data-raw '{
      "name":"test",
      "full_name": "test chef software"
    }'
    ```
   Response
   {}

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
